client/service/src/builder.rs

```diff
@@ -649,31 +650,41 @@ async fn transaction_notifications<TBl, TExPool>(
 		.await;
 }
 
-fn build_telemetry<TBl: BlockT>(
+fn init_telemetry(config: &Configuration) -> Option<sc_telemetry::Telemetry> {
+	let endpoints = match config.telemetry_endpoints.clone() {
+		Some(endpoints) if !endpoints.is_empty() => endpoints,
+		// we don't want the telemetry to be initialized if telemetry_endpoints == Some([])
```

Comment:
You need to know that this code was not introduced in this PR. It was introduced on #6666

I checked the code and I don't think sc-cli sends `Some([])`. It seems to sends the default of the chainspec if no argument is provided. But I think more investigation is probably out-of-scope with this PR (especially since this code was already there).

There is also one thing to consider: even if something was treating this endpoint array improperly, the type is `Option<Vec<_>>` so you need to handle the case `Some(vec![])` in case someone would use sc-service directly.